src/server/object_services/object_server.js

```diff
@@ -1415,6 +1417,19 @@ async function find_object_md(req) {
         obj = await MDStore.instance().find_object_latest(req.bucket._id, req.rpc_params.key);
     }
     check_object_mode(req, obj, 'NO_SUCH_OBJECT');
+
+    // fetch the object part range if a partNumber was requested
+    // and the object was uploaded using multipart upload
+    if (req.rpc_params.part_number && obj.num_parts) {
+        const parts = await MDStore.instance().find_part_of_object_by_number(obj, req.rpc_params.part_number);
```

Comment:
Hello Guy.

Thank you for your explanation, So S3 parts map into Noobaa multiparts and Noobaa multiparts could consist of one or more Noobaa de-dup parts. 

So before going into the woods of DB schema changes, fixed the calculation of the S3 part range for a specific part number using the existing MD data:

https://github.com/baum/noobaa-core/blob/75f549bd077eb1b62c9b2f32883a098aa2e48125/src/server/object_services/object_server.js#L1425-L1447

As for MDStore, added the following method, for multiparts less than equal given part number:

https://github.com/baum/noobaa-core/blob/75f549bd077eb1b62c9b2f32883a098aa2e48125/src/server/object_services/md_store.js#L917-L924

As for adding a start field for the multipart collection ( size if already there),  this might be not the most natural solution though. The offset within the object of the multipart is not known during the part uploaded, i.e. when a multipart is inserted into MDStore, however each multipart start could be calculated during multipart upload completion request. Then it is possible to go over all multiparts of the object and update each start field, does not sound cheap. So there is a trade off here.

What do you think?